nixos/modules/services/networking/infiniband.nix

```diff
@@ -0,0 +1,92 @@
+{lib, pkgs, config, ...}:
+
+let
+  cfg = config.networking.ib-interfaces;
+  systemd = config.systemd.package;
+in {
+
+  options = {
+    networking.ib-interfaces = {
+      enable = lib.mkEnableOption "ib-interfaces";
+      configDir = lib.mkOption {
+        default = "${pkgs.rdma-core}/etc/rdma/modules";
+        type = lib.types.path;
+        description = "The directory containing rdma kernel module configuration files";
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+
+    environment.systemPackages = [ pkgs.rdma-core ];
+    services.udev.packages =
+      let patchedUdevRules = pkgs.stdenv.mkDerivation {
+        name = "patched-${pkgs.rdma-core.name}-udev-rules";
+        src = "${pkgs.rdma-core}/lib/udev/rules.d";
+        phases = ["unpackPhase" "patchPhase" "installPhase"];
+        patchPhase = ''
+          substituteInPlace 60-srp_daemon.rules \
+            --replace /bin/systemctl ${systemd}/bin/systemctl
+        '';
+        installPhase = ''
+          mkdir -p $out/lib/udev/rules.d
+          cp -r . $out/lib/udev/rules.d/
+        '';
+      }; in [ patchedUdevRules ];
+
+
+    # This unit is reproduced from "${rdma-core}/lib/systemd/system/rdma-load-modules@.service";
+    systemd.services."rdma-load-modules@" = {
```

Comment:
I recall there were some bugs with template units. Did you verify this does the right thing?